tragedy strikes: --no-clean and no temp directory

## Expected Behavior

use with `--no-clean` flag doesn't throw if there's no temp dir (typically `.nyc_output`)

## Observed Behavior

```
Error: ENOENT: no such file or directory, scandir '/Users/boneskull/Documents/projects/mocha/.nyc_output'
    at Object.fs.readdirSync (fs.js:910:18)
    at NYC.loadReports (/Users/boneskull/Documents/projects/mocha/node_modules/nyc/index.js:479:31)
    at NYC._getCoverageMapFromAllCoverageFiles (/Users/boneskull/Documents/projects/mocha/node_modules/nyc/index.js:412:8)
    at NYC.report (/Users/boneskull/Documents/projects/mocha/node_modules/nyc/index.js:421:18)
    at report (/Users/boneskull/Documents/projects/mocha/node_modules/nyc/bin/nyc.js:89:7)
    at /Users/boneskull/Documents/projects/mocha/node_modules/nyc/bin/nyc.js:76:27
    at ChildProcess.<anonymous> (/Users/boneskull/Documents/projects/mocha/node_modules/nyc/node_modules/foreground-child/index.js:52:5)
    at emitTwo (events.js:125:13)
    at ChildProcess.emit (events.js:213:7)
    at maybeClose (internal/child_process.js:927:16)
make: *** [test-bdd] Error 1                                    
```

## Bonus Points! Code (or Repository) that Reproduces Issue

Well, `createTempDirectory()` [isn't called](https://github.com/boneskull/nyc/blob/master/index.js#L337) if [`clean` is falsy](https://github.com/boneskull/nyc/blob/master/bin/nyc.js#L41).

## Forensic Information

macOS; https://gist.github.com/boneskull/fbb28823e66a439f4a7fc735cb883917

--- a/bin/nyc.js
+++ b/bin/nyc.js
@@ -38,7 +38,11 @@ if (argv._[0] === 'report') {
   else config.instrumenter = './lib/instrumenters/istanbul'
 
   var nyc = (new NYC(config))
-  if (config.clean) nyc.reset()
+  if (config.clean) {
+    nyc.reset()
+  } else {
+    nyc.createTempDirectory()
+  }
   if (config.all) nyc.addAllFiles()
 
   var env = {

